eppo_metrics_sync/dbt_model_parser.py

```diff
@@ -0,0 +1,128 @@
+from collections import Counter
+from itertools import chain
+
+
+class DbtModelParser():
+
+    def __init__(self, model, dbt_model_prefix):
+        self.model = model
+        self.eppo_entities = []
+        self.eppo_timestamp = []
+        self.eppo_facts = []
+        self.eppo_properties = []
+        self.dbt_model_prefix = dbt_model_prefix
+    
+    def parse_columns(self):
+
+        for column in self.model.get('columns'):
```

Comment:
One idea here is to move the logic for one column to a separate method called something like `self._parse_one_column`. This makes it so the `parse_columns` function is now very simple and readable:

```
def parse_columns(self):
    for column in self.model['columns']:
        self._parse_one_column(column)  
```

Then the `self._parse_one_column` method is also easier to understand because its title clearly explains what all that logic does (it conveys intent)